### PR TITLE
solution to hw7-1

### PR DESCRIPTION
**Pull request by ggazila**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,53 +1,128 @@
...
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] elementsData;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: we need only one array for this task

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,53 +1,128 @@
...
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Object[] grow() {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: this method should return `void`

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,53 +1,128 @@
...
+        --size;
+        return oldValue;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        int i = 0;
+        if (t == null) {
+            for (; i < size; i++) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: please, simplify. one loop is just enough :)

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -1,53 +1,128 @@
...
+            elementsData = grow();
+        }
+        System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
+        elementsData[index] = value;
+        size += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: redundant

File: src/main/java/core/basesyntax/ArrayList.java [hunk 5]
```diff
@@ -1,53 +1,128 @@
...
+            return;
+        }
+        for (int i = 0; i < list.size(); i++) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: foreach might look better :)

File: src/main/java/core/basesyntax/ArrayList.java [hunk 6]
```diff
@@ -1,53 +1,128 @@
...
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Object[] grow() {
+        int oldCapacity = elementsData.length;
+        if (oldCapacity > 0) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: redundant

File: src/main/java/core/basesyntax/ArrayList.java (hunk 5 above)
Comment (ggazila) [reply src/main/java/core/basesyntax/ArrayList.java]: I can't understand how to do it. May I get a hint?
fore

File: src/main/java/core/basesyntax/ArrayList.java (hunk 1 above)
Comment (ggazila) [reply src/main/java/core/basesyntax/ArrayList.java]: ok

File: src/main/java/core/basesyntax/ArrayList.java (hunk 2 above)
Comment (ggazila) [reply src/main/java/core/basesyntax/ArrayList.java]: ok

File: src/main/java/core/basesyntax/ArrayList.java (hunk 3 above)
Comment (ggazila) [reply src/main/java/core/basesyntax/ArrayList.java]: ok

File: src/main/java/core/basesyntax/ArrayList.java (hunk 4 above)
Comment (ggazila) [reply src/main/java/core/basesyntax/ArrayList.java]: ok

File: src/main/java/core/basesyntax/ArrayList.java (hunk 6 above)
Comment (ggazila) [reply src/main/java/core/basesyntax/ArrayList.java]: ok

File: src/main/java/core/basesyntax/ArrayList.java (hunk 5 above)
Comment (Tarabarnik) [reply src/main/java/core/basesyntax/ArrayList.java]: well, I had a try, and it seems to be not really nice. sorry, my bad :( your solution is good

**Pull request closed without merging**